File: src/dynamicBody/character/player/Player.java
package dynamicBody.character.player;

import design.RoomDesign;
import design.utilities.Pair;
import dynamicBody.character.player.inventory.Inventory;
import dynamicBody.character.player.movement.Movement;
import dynamicBody.character.player.movement.check.CheckPlayerMovement;
import dynamicBody.character.player.shoot.ShootingPlayer;
import levels.Level;
import org.newdawn.slick.Input;
import org.newdawn.slick.SlickException;
import org.newdawn.slick.Sound;

/**
 * An interface that presents all the methods needed or that can be used with regards to the player 
 */

public interface Player extends dynamicBody.character.Character {
	
	/**
	 * Method used to set player's position
	 * @param input, received from the keyboard
	 * @param level, player's current level 
	 * @throws SlickException 
	 */
	void setPosition(Input input, Level level) throws SlickException;
	
	/**
	 * Method used to set the level in which the player should go
	 * @param level, to set the new player's level
	 */
	void setLevel(int level);
	
	/**
	 * @return the player's current level 
	 */ 
	int getLevel();
	
	/**
	 *  @return player's bullet 
	 */
	ShootingPlayer getBullet();

	/**
	 * @return player's current room 
	 */
	RoomDesign getRoom();

	/**
	 * Methods used to set player's current room 
	 * @param room, the room where the player would be moved
	 */
	void setCurrentRoom(RoomDesign room);
	
	/**
	 * Method used to update player's position
	 * @param position, the coordinates of the player to be set 
	 */
	void transitionPos(Pair<Integer, Integer> position);

	/**
	 * @return player's speed 
	 */
	int getPlayerSpeed();

	/**
	 * @return player's check 
	 */
	CheckPlayerMovement getCheck();
	
	/**
	 * Method used to upgrade player's damage 
	 * @param upgrade, an int value to add to player's current damage
	 */
	void upgradeDamage(int upgrade);
	
	/**
	 * @return player's inventory 
	 */
	Inventory getInventory();

	/**
	 * Method used to upgrade player's speed 
	 * @param upgrade, an int value to be added to player's speed in order to move faster in the dungeon
	 */
	void upgradePlayerSpeed(int upgrade);

	/**
	 * @return player's rate of fire 
	 */
	int getRateOfFire();
	
	/**
	 * Method used to load all player's animations 
	 * @throws SlickException
	 */
	void loadAnimations() throws SlickException;
	
	/**
	 * Method used to check the presence of any enemy in the room
	 * If the room is empty, the doors will  open and the player will be able to change room
	 * @param clearRoom, the boolean value to check 
	 */
	void setClearRoom(boolean clearRoom);
	
	/**
	 * @return player's sound walking into the dungeon 
	 */
	Sound getBowShoot();

	/**
	 * Method used to upgrade player's rate of fire 
	 * @param upgrade, an int value used to change player's current rate of fire in order to shoot faster 
	 */
	void upgradeRateOfFire(int upgrade);

	/**
	 * Method used to upgrade player's max health 
	 * @param upgrade, an int value to be added to the player's max health
	 */
	void upgradeMaxHealth(int upgrade);
	
	/**
	 * @return player's current health 
	 */
	int getCurrentHealth();
	
	/**
	 * @return player's max health 
	 */
	int getMaxHealth();

	/**
	 * Method used to upgrade player's current health
	 * @param heal, int value to be added to player's current health
	 */
	void heal(int heal);
	
	/**
	 * Method used to get player's movement (only used in class PlayerAndBulletCreationTest)
	 * @return new current player's coordinates and his direction  
	 */
	Movement getMove();	
}

File: Fabbri_Lucia/Csharp_LuciaFabbri/Player.cs
using System;
using System.Collections.Generic;
using System.Text;
using dynamicBody.bullet;
using dynamicBody.character.health;
using dynamicBody.character.player.inventory;
using dynamicBody.character.player.movement;
using dynamicBody.character.player.shoot;
using dynamicBody.move;
using main.worldModel;

namespace dynamicBody.character.player
{
    /// <summary>
    /// Class that implements interface Player used to define all the aspects concerning the player
    /// </summary>
    public class Player : IPlayer
    {
        private int level;
        private bool clearRoom;
        private int playerSpeed;
        private int damage;
        private int rateOfFire;
        private ValueTuple<int, int> position;
        private Direction direction;
        private IMovement move;
        private IHealth health;
        private Inventory inventory;
        private IShootingPlayer bullet;
        private ISet<IBullet> roomBullets =  new HashSet<IBullet>();

        /// <summary>
        /// Default constructor
        /// </summary>
        /// <param name="pos"> player's position </param>
        /// <param name="dir" player's direction ></param>
        /// <param name="level"> player's starting level </param>
        public Player(ValueTuple<int, int> pos, Direction dir, int level)
        {
            this.position = pos;
            this.level = level;
            this.direction = dir;
            this.playerSpeed = 2;
            this.rateOfFire = 800;
            this.damage = 10;
            this.inventory = new Inventory(this);
            this.health = new Health(100);
            this.clearRoom = false;
        }

        public int GetCurrentHealth()
        {
            return this.health.GetCurrentHealth();
        }

        public int GetDamage()
        {
            return this.damage;
        }

        public Direction GetDirection()
        {
            return this.direction;
        }

        public IInventory GetInventory()
        {
            return this.inventory;
        }

        public int GetLevel()
        {
            return this.level;  
        }

        public int GetMaxHealth()
        {
            return this.health.GetMaxHealth();
        }

        public IMovement GetMove()
        {
            return this.move;
        }

        public int GetPlayerSpeed()
        {
            return this.playerSpeed;
        }

        public ValueTuple<int, int> GetPosition()
        {
            return this.position;
        }

        public int GetRateOfFire()
        {
            return this.rateOfFire;   
        }

        public ISet<IBullet> GetRoomBullets()
        {
            return this.roomBullets;
        }

        public IShootingPlayer GetShootingBullet()
        {
            return this.bullet;
        }

        public void Heal(int heal)
        {
            this.health.Heal(heal);
        }

        public bool IsAlive()
        {
            return this.health.IsAlive();
        }

        public void ResetStats()
        {
            this.playerSpeed = 2;
            this.rateOfFire = 800;
            this.damage = 10;
            this.inventory = new Inventory(this);
            this.health = new Health(100);
        }

        public void SetClearRoom(bool clearRoom)
        {
            this.clearRoom = clearRoom;
        }

        public void SetLevel(int level)
        {
            this.level = level;
        }

        public void TakeDamage(int damage)
        {
            this.health.TakeDmg(damage);
        }

        public void TransitionPos(ValueTuple<int, int> position)
        {
            this.position = position;
        }

        public void UpgradeDamage(int upgrade)
        {
            this.damage = this.damage + upgrade;
        }

        public void UpgradeMaxHealth(int upgrade)
        {
            this.health.UpgradeMaxHealth(upgrade);
        }

        public void UpgradePlayerSpeed(int upgrade)
        {
            this.playerSpeed = this.playerSpeed + upgrade;
        }

        public void UpgradeRateOfFire(int upgrade)
        {
            this.rateOfFire = this.rateOfFire - upgrade;
        }
    }
}